diff -rup mysql-5.6.19-orig/scripts/CMakeLists.txt mysql-5.6.19/scripts/CMakeLists.txt
--- mysql-5.6.19-orig/scripts/CMakeLists.txt	2014-07-22 10:07:12.053410263 +0200
+++ mysql-5.6.19/scripts/CMakeLists.txt	2014-07-22 10:10:44.102502369 +0200
@@ -381,6 +381,28 @@ ELSE()
     ENDIF()
     INSTALL_SCRIPT(${CMAKE_CURRENT_BINARY_DIR}/${file} COMPONENT ${${file}_COMPONENT})
   ENDFOREACH()
+
+  # files for systemd
+  SET(SYSTEMD_SCRIPTS
+    mysql.tmpfiles.d
+    mysqld.service
+    mysqld-prepare-db-dir
+    mysqld-wait-ready
+    mysqld-check-socket
+    mysqld-scripts-common
+  )
+  FOREACH(file ${SYSTEMD_SCRIPTS})
+    IF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${file}.sh)
+      CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/${file}.sh
+        ${CMAKE_CURRENT_BINARY_DIR}/${file} ESCAPE_QUOTES @ONLY)
+    ELSEIF(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${file}.in)
+      CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/${file}.in
+        ${CMAKE_CURRENT_BINARY_DIR}/${file} ESCAPE_QUOTES @ONLY)
+    ELSE()
+      MESSAGE(FATAL_ERROR "Can not find ${file}.sh or ${file}.in in "
+        "${CMAKE_CURRENT_SOURCE_DIR}" )
+    ENDIF()
+  ENDFOREACH()
 ENDIF()
 
 # Install libgcc as  mylibgcc.a
