diff -up mysql-5.5.31/mysql-test/lib/My/ConfigFactory.pm.p22 mysql-5.5.31/mysql-test/lib/My/ConfigFactory.pm
--- mysql-5.5.31/mysql-test/lib/My/ConfigFactory.pm.p22	2013-04-24 20:15:14.085623163 +0200
+++ mysql-5.5.31/mysql-test/lib/My/ConfigFactory.pm	2013-04-24 20:15:37.456630939 +0200
@@ -36,7 +36,7 @@ my @pre_rules=
 );
 
 
-my @share_locations= ("share/mysql", "sql/share", "share");
+my @share_locations= ("share/community-mysql", "sql/share", "share");
 
 
 sub get_basedir {
diff -up mysql-5.5.31/mysql-test/mysql-test-run.pl.p22 mysql-5.5.31/mysql-test/mysql-test-run.pl
--- mysql-5.5.31/mysql-test/mysql-test-run.pl.p22	2013-03-25 14:14:58.000000000 +0100
+++ mysql-5.5.31/mysql-test/mysql-test-run.pl	2013-04-24 20:18:28.538687866 +0200
@@ -1238,11 +1238,11 @@ sub command_line_setup {
   }
 
   # Look for language files and charsetsdir, use same share
-  $path_language=   mtr_path_exists("$bindir/share/mysql",
+  $path_language=   mtr_path_exists("$bindir/share/community-mysql",
                                     "$bindir/sql/share",
                                     "$bindir/share");
   my $path_share= $path_language;
-  $path_charsetsdir =   mtr_path_exists("$basedir/share/mysql/charsets",
+  $path_charsetsdir =   mtr_path_exists("$basedir/share/community-mysql/charsets",
                                     "$basedir/sql/share/charsets",
                                     "$basedir/share/charsets");
 
@@ -3324,7 +3324,7 @@ sub mysql_install_db {
   }
 
   my $path_sql= my_find_file($install_basedir,
-			     ["mysql", "sql/share", "share/mysql",
+			     ["mysql", "sql/share", "share/community-mysql",
 			      "share", "scripts"],
 			     "mysql_system_tables.sql",
 			     NOT_REQUIRED);
